lib/Runtime/Language/InlineCache.cpp

```diff
@@ -697,7 +674,7 @@ namespace Js
 #endif
             if (!PHASE_OFF1(Js::CloneCacheInCollisionPhase))
             {
-                if (!inlineCaches[inlineCacheIndex].IsEmpty() && !inlineCaches[inlineCacheIndex].NeedsToBeRegisteredForStoreFieldInvalidation())
+                if (!inlineCaches[inlineCacheIndex].IsEmpty() && !inlineCaches[inlineCacheIndex].NeedsToBeRegisteredForStoreFieldInvalidation() && GetSize() != 1)
```

Comment:
Ok, makes sense.

---
In reply to: [115339004](https://github.com/Microsoft/ChakraCore/pull/2883#discussion_r115339004) [](ancestors = 115339004)